js/generate.js

```diff
@@ -66,22 +67,38 @@ function maturityData(spec) {
 	maturity = {label: spec.maturity, level:level};
     } else {
 	maturity = {label:spec.maturity, level: maturityLevels[spec.maturity]};
-	maturityIcon = {src:"http://www.w3.org/2013/09/wpd-rectrack-icons/" + spec.maturity.toLowerCase().replace(/lastcall/,'lcwd') + '.svg', alt:spec.maturity, width:50, height:50};
-	if (spec.maturity == "REC" || spec.maturity == "LastCall") {
+	maturityIcon = {src:"http://www.w3.org/2013/09/wpd-rectrack-icons/" + maturityMapping[spec.maturity].toLowerCase() + '.svg', alt:spec.maturity, width:50, height:50};
+	if (spec.maturity == "Recommendation" || spec.maturity == "Last Call") {
 	    maturityIcon.height = 53;
 	}
     }
     return {maturity: maturity, maturityIcon: maturityIcon};
 }
 
-var specData;
-var specXhr = new XMLHttpRequest();
-specXhr.open("GET", "specs/tr.json");
-specXhr.onload = function() {
-    specData = JSON.parse(this.responseText);
-    fillTables();
-};
-specXhr.send();
+var specData = {};
+fillTables();
+var w3cApiAuth = "?apikey=rvl38z1hpvkw0g804gk48o0s8408w88";
+var w3cApiFetchConfig = { method:'GET', mode:'cors'};
+
+
+function fetchSpecData(d) {
```

Comment:
Suggestion, you could destructure `d` maybe? It would make it more clear:
```JS
function fetchSpecData({maturity, wgs, title, TR} = {}) {
    if (!TR) {
        return new Promise((res,rej) => { res({maturity, wgs, title});});
    }
```